main.go

```diff
@@ -159,10 +158,12 @@ func main() {
 	}
 
 	// Add readiness probe
-	if err := mgr.AddReadyzCheck("readyz", healthz.Ping); err != nil {
+	if err := mgr.AddReadyzCheck("readyz", storagecluster.ReadinessChecker); err != nil {
 		setupLog.Error(err, "unable add a readiness check")
 		os.Exit(1)
 	}
+	storagecluster.ReadinessSet()
+	defer storagecluster.ReadinessUnset()
```

Comment:
Hmm... I'm not sure this is entirely necessary. Looking to the end of `main()`, the only thing that could happen is that `mgr.Start()` exits, at which point the container process would stop and the Pod would restart.

Reply:
That's a good point, calling ReadinessUnset() when the program is closing isn't actually going to do anything since at that point the http server used as the readiness probe will already be down.